GreysonElkins/abstractify
# Flatten data

#### What's this PR do?
* Restructures some of the data in `App.state`
* Prevents new fetches from wiping the data from old ones
* Changes the lock hover styling
* Attempts to control the location of rendered images that are locked
#### Where should the reviewer start?
* `MainPage` has a sort method which isn't behaving the way I'd expect (#27)
* Follow data from `componentDidMount` to `App.state.savedSets` 
#### How should this be manually tested?
Well as soon as you start locking images and refreshing you'll see some different behavior.
#### What are the relevant tickets?
close #35 #9 #27
#### Questions:
Did you know you could accidentally tag users in commit messages @27? Sorry!
